### PR TITLE
[#738]:Dependencies follow alphabetical order

### PR DESCRIPTION
**Pull request by hiirrxnn**
### What changes were proposed in this pull request?

Dependencies follow alphabetical order

### Why are the changes needed?
For better readability
Fix: #738 
### Does this PR introduce _any_ user-facing change?
N/A
### How was this patch tested?

N/A

### CODE REVIEW

**Comment by hiirrxnn** [comment]
Please take a look , if this looks good , I will do the same for all subsequent build.gradle.kts files . This is just a demo.

**Comment by justinmclean** [comment]
Look good, but I think you may have moved a little too much around. It's just the content inside the dependencies block that needs to change.

**Comment by hiirrxnn** [comment]
> Look good, but I think you may have moved a little too much around. It's just the content inside the dependencies block that needs to change.

Could you review in the files changed where I changed something I shouldn't have ,so that its clear. Should I not rearrange the 'exclude(xx)' dependencies.

File: catalogs/catalog-lakehouse-iceberg/build.gradle.kts [hunk 1]
```diff
@@ -11,69 +11,66 @@ plugins {
...
+  implementation(libs.guava)
   implementation(libs.iceberg.hive.metastore)
+  implementation(libs.jackson.annotations)
+  implementation(libs.jackson.databind)
+  implementation(libs.jackson.datatype.jdk8)
+  implementation(libs.jackson.datatype.jsr310)
   implementation(libs.sqlite.jdbc)
+  implementation(project(":api"))
+  implementation(project(":common"))
+  implementation(project(":core"))
```
Comment (jerryshao) [inline catalogs/catalog-lakehouse-iceberg/build.gradle.kts:30]: These are always on the top.

File: catalogs/catalog-lakehouse-iceberg/build.gradle.kts [hunk 2]
```diff
@@ -11,69 +11,66 @@ plugins {
...
   testImplementation(libs.jersey.test.framework.core) {
     exclude(group = "org.junit.jupiter")
   }
   testImplementation(libs.jersey.test.framework.provider.jetty) {
     exclude(group = "org.junit.jupiter")
   }
+  testImplementation(libs.junit.jupiter.api)
+  testImplementation(libs.junit.jupiter.params)
+  testImplementation(libs.mockito.core)
+  testRuntimeOnly(libs.junit.jupiter.engine)
```
Comment (jerryshao) [inline catalogs/catalog-lakehouse-iceberg/build.gradle.kts:73]: Please group them separately. For example, like:

```
implementation(xxx)

annotationProcessor(xxx)

compileOnly(xxx)

testImplementation(xxx)

testRuntimeOnly(xxx)

```

**Comment by yuqi1129** [comment]
@hiirrxnn Do you need help?  I suggest you run "./gradlew build -DskipTests -DskipITs" before pushing code to GitHub to ensure everything is OK locally.

**Comment by hiirrxnn** [comment]
I have run the said command in check details but the issue persists.

**Comment by yuqi1129** [comment]
@hiirrxnn 
You still have some fomat issues. can you provide your local environment:
- IDE you use and related-configuration like space and tabs as follows
image
 
- JDK version 
- Local OS
- and so on.

**Comment by hiirrxnn** [comment]
I'm currently using VS Code IDE , JDK version : openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+10 (build 17.0.6+10)
OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (build 17.0.6+10, mixed mode) , and MacOS Sonoma 14.2.1 operating system

**Comment by hiirrxnn** [comment]
Should i use a specific IDE or formatting conditions ? Please let me know

**Comment by yuqi1129** [comment]
> Should i use a specific IDE or formatting conditions ? Please let me know

No, I don't think the project should be limited to a specific IDE or formatting settings. I noticed that CI is now okay.

**Review by yuqi1129 — APPROVED** [review]
It's okay for me.

**Comment by hiirrxnn** [comment]
I will commit/push each edited file , so that it's easy to pinpoint which one is causing the error.

File: catalogs/bundled-catalog/build.gradle.kts [hunk 3]
```diff
@@ -11,12 +11,12 @@ plugins {
 }
 
 dependencies {
-  implementation(project(":core"))
+  implementation(libs.slf4j.api)
```
Comment (qqqttt123) [inline catalogs/bundled-catalog/build.gradle.kts:14]: The section of `lib` should be bottom.

**Comment by hiirrxnn** [comment]
This push is a draft , I will review it if any CI fails.

**Comment by justinmclean** [comment]
You need to run `./gradlew :core:spotlessKotlinGradleCheck`

**Comment by hiirrxnn** [comment]
Build is still failing

**Comment by justinmclean** [comment]
If you look at the details you'll see it will tell you where it is failing in this case, you need to run `./gradlew :server:spotlessApply`, running `./gradlew build -x test` locally would also give you the same error, so it's best to do that first before pushing your changes.

**Comment by yuqi1129** [comment]
> CI is still failing .Could you help me pinpoint which specific files are causing the problem ?

Here is the patch. Please take a look. 
[patch.txt](https://github.com/datastrato/gravitino/files/14133613/patch.txt)

**Comment by jerryshao** [comment]
Thanks @hiirrxnn for the PR. @yuqi1129 @qqqttt123 do you have further comments?

**Comment by yuqi1129** [comment]
> Thanks @hiirrxnn for the PR. @yuqi1129 @qqqttt123 do you have further comments?

I'm okay with the changes.

**Comment by hiirrxnn** [comment]
I still have one file remaining to sort , since because of it CI was failing , I pushed it untouched and will edit it once you guys approve the current changes Let me know if there are other improvements to be made.

**Comment by yuqi1129** [comment]
> I still have one file remaining to sort , since because of it CI was failing , I pushed it untouched and will edit it once you guys approve the current changes Let me know if there are other improvements to be made.

Hi, Is your local environment Windows? Can you help try this
https://github.com/datastrato/gravitino/pull/1976#discussion_r1477033104, Still we haven't tested the spotless plugin with Windows comprehensively

**Comment by hiirrxnn** [comment]
I'm sorry but I use MacOS.

**Comment by yuqi1129** [comment]
@hiirrxnn
Could you please resolve the conflicts?

**Comment by hiirrxnn** [comment]
Sure

File: api/build.gradle.kts [hunk 4]
```diff
@@ -9,14 +9,14 @@ plugins {
...
+  implementation(libs.commons.lang3)
   implementation(libs.guava)
   implementation(libs.slf4j.api)
-  implementation(libs.commons.lang3)
 
   testImplementation(libs.junit.jupiter.api)
   testImplementation(libs.junit.jupiter.params)
-
   testRuntimeOnly(libs.junit.jupiter.engine)
+  testImplementation(libs.junit.jupiter.params)
```
Comment (yuqi1129) [inline api/build.gradle.kts:19]: Why do we introduce a duplicated dependency `libs.junit.jupiter.params`?

File: integration-test/build.gradle.kts [hunk 5]
```diff
@@ -73,62 +74,58 @@ dependencies {
...
-  testImplementation(libs.sqlite.jdbc)
+  testImplementation(libs.selenium)
+  testImplementation(libs.slf4j.jdk14)
   testImplementation(libs.spark.hive)
+  testImplementation(libs.sqlite.jdbc)
   testImplementation(libs.testcontainers)
   testImplementation(libs.testcontainers.junit.jupiter)
   testImplementation(libs.testcontainers.mysql)
   testImplementation(libs.testcontainers.postgresql)
-  testImplementation(libs.trino.jdbc)
```
Comment (yuqi1129) [inline integration-test/build.gradle.kts:115]: Why this is missing?

Comment (yuqi1129) [reply integration-test/build.gradle.kts:115]: @hiirrxnn 
Please undo the changes and resolve any conflicts.

**Review by yuqi1129 — APPROVED** [review]
LTGM, @qqqttt123 do you have any further comments?

**Comment by yuqi1129** [comment]
@hiirrxnn
Thank you for your contributions.

**Pull request merged**